mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAAttachmentMapper.java

```diff
@@ -0,0 +1,154 @@
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.jpa.mail;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import org.apache.james.mailbox.exception.AttachmentNotFoundException;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.jpa.JPAId;
+import org.apache.james.mailbox.jpa.JPATransactionalMapper;
+import org.apache.james.mailbox.jpa.mail.model.JPAAttachment;
+import org.apache.james.mailbox.model.Attachment;
+import org.apache.james.mailbox.model.AttachmentId;
+import org.apache.james.mailbox.model.MessageId;
+import org.apache.james.mailbox.store.mail.AttachmentMapper;
+import org.apache.james.mailbox.store.mail.model.Username;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.PersistenceException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class JPAAttachmentMapper extends JPATransactionalMapper implements AttachmentMapper {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JPAAttachmentMapper.class);
+
+
+    public JPAAttachmentMapper(EntityManagerFactory entityManagerFactory) {
+        super(entityManagerFactory);
+    }
+
+
+    @Override
+    public Attachment getAttachment(AttachmentId attachmentId) throws AttachmentNotFoundException {
+        Preconditions.checkArgument(attachmentId != null);
+        return Optional.ofNullable(getEntityManager().find(JPAAttachment.class, attachmentId.getId()))
+                .orElseThrow(() -> new AttachmentNotFoundException(attachmentId.getId())).toAttachment();
+    }
+
+    @Override
+    public List<Attachment> getAttachments(Collection<AttachmentId> attachmentIds) {
+        Preconditions.checkArgument(attachmentIds != null);
+        return ImmutableList.<Attachment>builder().addAll(
```

Comment:
you can use `attachmentIds.stream().map()....collect(Guavate.toImmutableList())` to produce an immutable list. Without using the builder.